python/cudf/cudf/tests/test_string.py

```diff
@@ -2922,3 +2922,17 @@ def test_string_std():
     assert_exceptions_equal(
         lfunc=psr.std, rfunc=sr.std, compare_error_message=False
     )
+
+
+def test_string_slice_with_mask():
+    actual = cudf.Series(["hi", "hello", None])
+    expected = actual[0:3]
+
+    assert actual._column.base_size == 3
+    assert_eq(actual._column.base_size, expected._column.base_size)
+    assert_eq(actual._column.null_count, expected._column.null_count)
+    assert_eq(
+        actual._column.mask.to_host_array(),
+        expected._column.mask.to_host_array(),
+    )
+    assert_eq(actual, expected)
```

Comment:
Can we convert this to something that tests user-facing behaviour rather than internal behaviour?

In other words, did this bug manifest in a way that affected end-users? If so, can we test that we fixed that instead?

Reply:
Yeah I had this similar thought initially and thought we could check with `isnull` public API, but since this goes to libcudf call and that returns the correct result without interacting with `Column.mask` we cannot validate using `Series.isnull`. 

The closest user-facing behaviour where this issue would surface is when we round-trip(when it goes through `serialize`) a series with string dtype & having nulls like in `test_distributed::test_str_series_roundtrip`.

Reply:
Sounds like we should definitely add a test for that then (maybe in `test_serialize.py`?).

We can keep this test in addition, if you prefer. Personally, I'm not a fan of testing internals, but that could be just me :-)

Reply:
Added a serialization test where we would still have to validate an internal component(i.e., the frames) and removed checking for mask and retained `base_size` checks in `test_string.py`

Reply:
I don't understand -- what internal attribute are we testing in the new serialize test?

Reply:
Serialize returns a dict & Frames as buffers. The internal attribute we are testing here is Frames`([index_frame, offset_frame, chars_frame, mask_frame])`, to be specific we want to validate the `mask_frame` at last index which would be the right validation.

Reply:
Got it. Thanks!